supervisor/docker/manager.py

```diff
@@ -421,43 +530,61 @@ def run(
                 f"Dockerd connection issue for {name}: {err}", _LOGGER.error
             ) from err
 
-        # Attach network
-        if not network_mode:
-            alias = [hostname] if hostname else None
-            try:
-                self.network.attach_container(container, alias=alias, ipv4=ipv4)
-            except DockerError:
-                _LOGGER.warning("Can't attach %s to hassio-network!", name)
+        # Get container metadata
+        try:
+            container_attrs = await container.show()
+        except aiodocker.DockerError as err:
+            raise DockerAPIError(
+                f"Can't inspect new container {name}: {err}", _LOGGER.error
+            ) from err
+        except requests.RequestException as err:
+            raise DockerRequestError(
+                f"Dockerd connection issue for {name}: {err}", _LOGGER.error
+            ) from err
+
+        # Setup network and store container id in cidfile
+        def setup_network_cidfile() -> None:
```

Comment:
It feels a bit awkward it's done in a single function. At least we should call it `setup_network_and_cidfile`, but maybe rather just have a separate function that we'll pass to the executor. In the end they'll be blocking on different operations (Docker API vs. FS I/O) so it IMHO makes sense to have it separated.

Reply:
The problem is there's a lot of performance overhead simply delegating a task to the executor in python, to my understanding from talking with bdraco it's a pretty expensive operation. So doing this as two separate functions instead of one may look cleaner but has a non-negligible performance impact on the overall process. That's why I combined them here since in general we try to limit the number of tasks delegated to the executor when possible.

If we think its inconvenient enough I can try and benchmark it to see what the real cost is. But I feel like I'd prefer to leave it and just do the best we can with name and comments.